test/utils/wasm3_engine.cpp

```diff
@@ -38,7 +40,35 @@ class Wasm3Engine final : public WasmEngine
 
 namespace
 {
-const void* env_adler32(IM3Runtime /*runtime*/, uint64_t* stack, void* mem)
+M3ValueType translate_valtype(char input)
+{
+    if (input == 'i')
+        return M3ValueType::c_m3Type_i32;
+    else if (input == 'I')
+        return M3ValueType::c_m3Type_i64;
+    else
+        throw std::runtime_error{"invalid type"};
+}
+
+std::pair<std::vector<M3ValueType>, std::vector<M3ValueType>> translate_signature(
```

Comment:
It would be nice to make a generic helper for all engines, maybe like
```cpp
template<typename ValueType>
std::pair<std::vector<ValueType>, std::vector<ValueType>> translate_signature(std::string_view signature, ValueType i32_type, Value_Type i64_type);
```

Can be a separate PR.

Reply:
Addressed in #741.